### DISCUSSION TITLE
How to warn users about not supporting encrypted tokens in the Hosted-App

### DISCUSSION THREAD

**Discussion by nabeelsaabna** [renovatebot/renovate#28895] [Suggest an Idea]
### Tell us more.

In the developer.mend.io hosted app we aim to deprecated the use of encrypted tokens in repository config (e.g. `renovate.json`) and for that a added a the ability to save/migrate secrets in the UI (no more `privateKey`)

Suggested solution:
Add a new self-hosted config option `privateKeyWarning`  that will be printed once with `warn` level if a an encrypted token is found during the run

**Comment by rarkins** [comment]
Could be `privateKeyWarning` or `encryptedWarning`.

Ideally the warning message is displayed in each PR plus in the Dependency Dashboard

**Reply by nabeelsaabna** [reply]
`encryptedWarning` sounds better, specially that there are `privateKey` and `privateKeyOld`

**Reply by rarkins** [reply]
I'd like:
- A new page on docs.renovatebot.com explaining our preference for UI-uploaded secrets, and a section on how to migrate
- A warning message printed once per repo/job, which says something like "Support for encrypted secrets in config is deprecated by the Mend Renovate App" plus a link to this documentation page
- The warning is shown on dependency dashboard, and in indicator is shown in each PR mentioning that warnings were raised (not sure if we need the full message or not)

**Reply by PhilipAbed** [reply]
Should i make it a global config only? `GlobalConfig.get('encryptedWarning')`

**Reply by PhilipAbed** [reply]
I'm having a bit of trouble implementing it on every PR
1. configuration is decrypted and validated early. i can do a print there, and it appears in the dashboard in `repoProblems`.
2. the warnings for every PR come from branch config/lookup warnings, by the time i reach there the 'encrypted' key is already deleted. so i need to send an indication in the config that 'encrypted' was detected + the flag is on, but anyway i think of doing this seems messy.
i'm able to use the `extractRepoProblems` only without adding anything.

do we need this warning to appear on every PR?

**Reply by PhilipAbed** [reply]
i have tried to do it anyway
and i noticed that we have 2 Warnings sections, 1 for config warnings, the other for dependency lookup warnings
The other warning under the yellow Warning is for dependency lookup.
the flag is set to `encryptedWarning: type something here` see:
and the `Configuration Warning` is the topic of the warning i hardcoded
![image](https://github.com/renovatebot/renovate/assets/42116482/995ba441-ab89-4306-92e0-48feda355375)

**Reply by PhilipAbed** [reply]
1) i can either show the Full Message with Topic + Message like you see in the picture.
2) show the message under the yellow warning
3) just write under the `Warning` in yellow, ask the user to look for errors in the dashboard

**Reply by justo-mend** [reply]
Text to show on the PR: "Warnings were raised when the job ran. See the Dashboard issue for more details."

**Reply by PhilipAbed** [reply]
in order to be consistent with the above message(See screenshot), 
i used:
`Warnings were raised. Check the logs or Dashboard for more information.`
